### DISCUSSION TITLE
Magic Link & CDN issues

### DISCUSSION THREAD

**Discussion by alfainfotech** [mozilla/hubs#5157] [Hubs Cloud Questions]
Hi,

#1 Magic Link
Recently I started this magic link issue, no longer getting ADMIN or Non-Admin emails thru magic link. I tried to updated the STACK manually on AWS as instructed on documentation still same issue. No console error on browser. Please let me know what can be done. 

After updating STACK - (changing small  <--> medium) etc., now I'm getting this error when I try to sign-in.
phoenix.js:844 WebSocket connection to 'wss://jolly-dwarf.domainname.com/socket/websocket?vsn=2.0.0' failed: 
phoenix-utils.js:163 Socket error, changed endpoint to wss://jolly-dwarf.domainname.com/socket/websocket

#2 CDN  ClouldFlare

<img width="737" alt="Screen Shot 2022-02-17 at 10 38 55 PM" src="https://user-images.githubusercontent.com/1449569/154618867-8e4af292-5775-4edd-a323-33a820ef477e.png">

**Comment by pattersonbl2** [comment]
for these SES issues,  can you run through these steps here to send a test email?   https://docs.aws.amazon.com/ses/latest/dg/send-an-email-from-console.html.   I want to make sure the service is still working.  if this test email sends the server config has some errors. This mostly can be fixed by terminating the server and letting the auto-scaling bring up another one.

**Reply by pattersonbl2** [reply]
You could also start here with the CDN issue  https://github.com/mozilla/hubs/discussions/3732.

**Reply by alfainfotech** [reply]
> for these SES issues, can you run through these steps here to send a test email? https://docs.aws.amazon.com/ses/latest/dg/send-an-email-from-console.html. I want to make sure the service is still working. if this test email sends the server config has some errors. This mostly can be fixed by terminating the server and letting the auto-scaling bring up another one.

Yes, its working fine. Even after installing HC. I get the ADMIN activation email after that 1st time as ADMIN login I get email too. And then after no email for ADMIN and non-admin too. May be magic link has to do something with web socket error
 
phoenix.js:844 WebSocket connection to 'wss://nervous-troll.xxxxxxcom/socket/websocket?vsn=2.0.0' failed: 
phoenix-utils.js:163 Socket error, changed endpoint to wss://nervous-troll.xxxxxx.com/socket/websocket

**Reply by pattersonbl2** [reply]
I believe it does when I tried to test myself that same socket came up but mine worked fine.   Are you on a custom client?

**Reply by alfainfotech** [reply]
I'm using that AWS Personal setup widget.

**Reply by pattersonbl2** [reply]
Hmmm, what happens when you terminate the instance?  and what version are you on?

**Reply by alfainfotech** [reply]
Ok I deleted and reinstalled from fresh 12th time. Fresh HC Personal install and Admin email showed. Configured default avatar/scene. Homepage loading perfect.  As normal user - used another browser to login... NO EMAIL.... (not even in spam folder).
Used same browser and provided Admin email sign... EMAIL showed in the inbox. 

FYI: Admin email is different from normal user email.

**Reply by alfainfotech** [reply]
ATleast this install fixed the CDN (Cloudflare issue)... left with Email for non-admins issue.

**Reply by pattersonbl2** [reply]
Since you are able to get the admin email, you can ssh into your instance and run a journalctl to see what could be happening when you send an email to normal users.

**Reply by alfainfotech** [reply]
> Since you are able to get the admin email, you can ssh into your instance and run a journalctl to see what could be happening when you send an email to normal users.
I dont see any error I the journalctl command.. here is the last log lines when I clicked sign and enter email and click submit.. no email came to inbox.

**Reply by alfainfotech** [reply]
<img width="1152" alt="Screen Shot 2022-03-10 at 11 48 43 PM" src="https://user-images.githubusercontent.com/1449569/157810179-aa73bc62-417f-400c-82d6-d4906f624274.png">
